### ISSUE TITLE
details_url different of theme URI hosted on GitHub

### ISSUE THREAD

**Issue by wolfgangdie** [YahnisElsts/plugin-update-checker#329]
Hi! I am implementing PUC v4.8.1 for a theme hosted on GitHub.

Everything works perfectly, but I would like to know how if I could manually configure the `details_url` value, because by default on GitHub, it acquires the value of the URI theme, but I would like to be able to redirect to a link with the changelog when "View version x details" link appears in the update notification theme panel.

I tried using something similar that I managed to read in the repo issues to implement the images for a plugin hosted on GitHub, but I had no results:

```php
// Additional data for theme
$updater->addResultFilter(function($themeInfo, $httpResponse = null) {
    $themeInfo->details_url = 'http://example.com/changelog';
    return $themeInfo;
});
```

Thank you very much for any help you can give me, and for such an excellent job you have done with all this solution!

**Comment by YahnisElsts** [comment]
That code looks correct to me. I tried it with one of my test themes and it successfully changed the page that's displayed in the "view version x details" pop-up.

Could it be a caching issue? For themes, the result filter only runs when checking for updates which happens about every 12 hours by default. So if you add the filter to an active theme. you might not see an immediate change because it's still using the cached `details_url`. To refresh the update information, you'll need to make WordPress/PUC check for updates. You can do that through the [Debug Bar](https://wordpress.org/plugins/debug-bar/) or by visiting *Dashboard -> Updates*. However, refreshing the "Updates" page is not 100% reliable because it doesn't always trigger an immediate check, it just reduces the check interval.

**Comment by wolfgangdie** [comment]
Hi! You were absolutely right, I tried again by implementing the code and I did the cleaning of the cache with the [WP Optimize](https://wordpress.org/plugins/wp-optimize/) plugin, and later, I reduced the theme in one version, in that way, I got the update notification to appear again, and it started working properly as I checked the link.

Thank you so much for all the help! I wrote down everything I did in case it could be of help to someone else, and the issue can be closed too!
